src/SFA.DAS.Commitments.Infrastructure/Services/ApprenticeshipEventsPublisher.cs

```diff
@@ -26,9 +27,22 @@ public ApprenticeshipEventsPublisher(IEventsApi eventsApi, ICommitmentsLogger lo
 
         public async Task Publish(IApprenticeshipEventsList events)
         {
+            _logger.Info($"Publishing {events.Events.Count} events");
             var apiEvents = events.Events.Select(x => CreateEvent(x.Commitment, x.Apprenticeship, x.Event, x.EffectiveFrom, x.EffectiveTo, x.PriceHistory));
-            await _eventsApi.BulkCreateApprenticeshipEvent(apiEvents.ToList());
-            events.Clear();
+
+            var batches = SplitList(apiEvents.ToList(), _maxBatchSize).ToList();
+
+            if (batches.Count() > 1)
+            {
+                _logger.Info($"Splitting events into {batches.Count} batches of up to {_maxBatchSize}");
+            }
+
+            foreach (var batch in batches)
+            {
+                _logger.Info($"Calling events api to bulk create {batch.Count} events");
+                await _eventsApi.BulkCreateApprenticeshipEvent(batch.ToList());
+                events.Clear();
```

Comment:
why this call here? should it be outside the foreach? or are the events affected by the api.bulk...() call?